Commonalities/documentation/Working/user-consent.md

```diff
@@ -0,0 +1,405 @@
+
+Then the Marketplace polls the token endpoint by making an "HTTP POST" request by sending the grant_type (`urn:openid:params:grant-type:ciba`) and `auth_req_id` parameters (step 7). 
+Because the user accepted the Terms and Conditions previously, the authorization/authentication was completed, and it will return a HTTP 200 OK response to the Marketplace with an access token in the response body (step 8). 
+
+Now the Marketplace has a valid Access Token that can be used to invoke the network API.
+
+## Consent API and Consent Config API
+
+As described in previous sections, an Operator NaaS Platform shall be built following a privacy-by-default approach, which guarantees an application or service can only access or modify the personal information it has permission to. This means that every time an application wants to use an API containing personal information, the platform checks automatically the purpose, the purpose level, and the user consent. But this way of working requires to implement a Consent API.
+
+**Consent API** is required for two main reasons:
+
+1.	A third-party can register the consent gathered from a user for a specific purpose which grants access to user personal information associated to some specific scope/s. Consent is collected by a third-party and not directly managed in auth server.
+2.	The auth server can validate during delegated authorization process if there is user consent (opt-in) registered for the user and the requested purpose/s as pre-requisite to issue an access token.
+ 
+Operator NaaS platform MUST define **Consent API** to meet the requirement where user consents are gathered and managed by a third-party element and not by Auth server itself.      
+
+The API allows an application back-end to perform the following actions on behalf of a specific user:
+
+•	Find if it has the consent to fulfill a certain purpose.
+•	Create a consent
+•	Revoke a consent
+•	Simply inform that the consent was prompted to the user but he did not accept it.
+
+But there is another API that may be required by third parties, called **Consent Config API**. This API would be intended for the management and provision of purposes, legal entities, and PI scopes. In order to request access to Personal Information under certain purpose, that purpose must be previously provisioned (see example in [Purposes](#purposes) section), and the PI scopes associated properly defined. In the same way, user consent is created for the legal entity associated to the client application and those legal entities must be also provisioned.
+
+## Annex: Flow source code
+
+### Obtaining an access token to manage user consent flow
+
+```
+title Obtaining Tokens to Manage Consents with CIBA (delegated consent)
+  
+participant User as User   
+participant App FE as FE      
+participant App BE as BE    
+participant "Marketplace" as MP     
+participant "Operator NaaS Platform" as ONP    
+  note over User, FE: T&C presented to User     
```

Comment:
Is there any provision in this to guarantee that the AppFE _actually_ displays the T&C verbatim?